flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/AbstractStreamOperatorV2.java

```diff
@@ -93,8 +93,8 @@
 	protected final LatencyStats latencyStats;
 	protected final ProcessingTimeService processingTimeService;
 
-	private StreamOperatorStateHandler stateHandler;
-	private InternalTimeServiceManager<?> timeServiceManager;
+	protected StreamOperatorStateHandler stateHandler;
+	protected InternalTimeServiceManager<?> timeServiceManager;
```

Comment:
Could we somehow limit the visibility of those fields? Maybe `package-private`? Or at the very least mark it as `@Internal`?

Reply:
Thanks for the suggestion. I find `AbstractStreamOperatorV2` has already provided a protected method named `getTimeServiceManager`. We can also add a `protected` method named `getStateHandler`.  So that its sub-classes can access `stateHandler` and `timeServiceManager` both. I will revert the previous change.